WFP-VAM/prism-app
# Get start and end date for validity period from admin level data; COUNTRY=rbd

This fixes issue #751.

How to test the feature:

1. Use RBD country with phase layer.
2. Timeline should load with a direction for each start and end date for every periode range 

Notes:
- every `admin_level_data` layer type having a `path` also have `start_date` and `end_date` fields in there corresponding data files. 
- it's always  `start_date` and `end_date`, hence it does not need to be configurable

**TODO**
- [x] POC on phase classification for rbd 
- [x] Ensure working only when a path is defined for the layer and start_date end_date is defined
- [x] Remove unused validity object for layer that now use start_date end_date feature 
- [x] Safe http request on data ressources (~~timeout~~, error, ...)
- [x] Draft another ticket for perf improvements => https://github.com/WFP-VAM/prism-app/issues/885
- [x] Draft another ticket to generate DateItem only for visible layers and on the fly instead of all layers at application first load => https://github.com/WFP-VAM/prism-app/issues/888

Screenshot/video of feature:

Before (RBD, Population in phase 3 to 5)
![image](https://github.com/WFP-VAM/prism-app/assets/100351051/297f14b7-7c55-43e9-bc5e-10045eb5975c)


Now :
![image](https://github.com/WFP-VAM/prism-app/assets/100351051/18ee2bc4-f9a4-4d0d-86a5-75fe9bfc2f8e)

With rainfall
![image](https://github.com/WFP-VAM/prism-app/assets/100351051/76b175fc-c8e3-4758-b34c-9615f8a8ca95)

Test URL :  https://prism-877.surge.sh/?baselineLayerId=ch_pop3_5&date=2023-06-01&hazardLayerIds=rainfall_dekad
